### PR TITLE
Ability to open a new Tab in current iTerm window

### PR DESCRIPTION
**Pull request by vitorhsb**
A new parameter can be used to instruct to open a new tab in the current window or to open a new window altogether. Also, a few bug corrections on the RUNNING variable.

### CODE REVIEW

**Comment by jcrben** [comment]
I copied this into my file and it works great for me (changed "iTerm" to "iTerm 2") - nice work.

File: iTerm.sh [hunk 1]
```diff
@@ -2,11 +2,22 @@
...
+												;;
+	esac
+	shift
+done
 
 if (( $(expr $VERSION '<' 10.7) )); then
 	RUNNING=$(osascript<<END
 	tell application "System Events"
-	    count(processes whose name is "iTerm")
+			count(processes whose name is "iTerm")
```
Comment (twolfson) [inline iTerm.sh]: Why is there excess whitespace here?

**Comment by twolfson** [comment]
To make sure we are on the same page, the intended usage is to use the following settings:

```json
{
  "terminal": "iTerm.sh",
  "parameters": ["--openNewTab"]
}
```

File: iTerm.sh (hunk 1 above)
Comment (vitorhsb) [inline iTerm.sh]: Sorry about the delay. There's no need for the extra whitespace, you can safely delete it.

Comment (twolfson) [inline iTerm.sh]: Could you update the PR to reflect the same indentation as the original file?

**Comment by twolfson** [comment]
@vitorhsb You still haven't replied to my last question on the main thread =/

**Comment by vitorhsb** [comment]
@twolfson A new commit has been made to remove the extra whitespace and reflect your concerns about intended usage. By default, the command will open a new terminal window, but can be changed to tab with the parameter --openNewTab .

```
{
  "terminal": "iTerm.sh",
  "parameters": ["--openNewTab"]
}
```

File: iTerm.sh [hunk 2]
```diff
@@ -2,11 +2,21 @@
 
 CD_CMD="cd "\\\"$(pwd)\\\"" && clear"
 VERSION=$(sw_vers -productVersion)
+NEWTAB=0
+
+while [ "$1" != "" ]; do
```
Comment (twolfson) [inline iTerm.sh]: Please use `((` for expressions to make this consistent with the rest of the file

File: iTerm.sh [hunk 3]
```diff
@@ -2,11 +2,21 @@
...
+			;;
+	esac
+	shift
+done
 
 if (( $(expr $VERSION '<' 10.7) )); then
 	RUNNING=$(osascript<<END
 	tell application "System Events"
-	    count(processes whose name is "iTerm")
+		count(processes whose name is "iTerm")
```
Comment (twolfson) [inline iTerm.sh]: There is still unnecessary whitespace noise here...

File: iTerm.sh [hunk 4]
```diff
@@ -2,11 +2,21 @@
 
 CD_CMD="cd "\\\"$(pwd)\\\"" && clear"
 VERSION=$(sw_vers -productVersion)
+NEWTAB=0
+
+while [ "$1" != "" ]; do
+	case $1 in
+		--openNewTab)
```
Comment (twolfson) [inline iTerm.sh]: Typically CLI utilities are `dash-cased`, can we update this to use dash-case as well?

```
--open-new-tab
```

File: iTerm.sh (hunk 3 above)
Comment (vitorhsb) [inline iTerm.sh]: Sorry, I'm not getting where the unnecessary whitespace is. The difference you can see here is changing spaces in the original file for a tab.

File: iTerm.sh (hunk 2 above)
Comment (vitorhsb) [inline iTerm.sh]: I'm not an expert in bash scripts, but I believe that this is the correct construct.

File: iTerm.sh (hunk 4 above)
Comment (vitorhsb) [inline iTerm.sh]: I believe we can simplify it a little more. Changed for 
```
-t 
```
or
```
--tab
```

File: iTerm.sh (hunk 3 above)
Comment (twolfson) [inline iTerm.sh]: Right, what I am trying to get at is that we don't need that change in this PR. It is creating unnecessary noise. If it's a matter of making the file more consistent, please do that in another PR focused on cleaning up inconsistencies.

File: iTerm.sh (hunk 2 above)
Comment (twolfson) [inline iTerm.sh]: There are many ways to perform a conditional expression in `bash` (e.g. `test`, `[`, `[[`). I overextended my assumptions due to an improper usage of `((` later on in this file. You are correct, `((` is reserved for arithmetic evalution:

http://linux.die.net/man/1/bash

File: iTerm.sh [hunk 5]
```diff
@@ -2,11 +2,23 @@
...
 CD_CMD="cd "\\\"$(pwd)\\\"" && clear"
 VERSION=$(sw_vers -productVersion)
+NEWTAB=0
+
+while [ "$1" != "" ]; do
+	PARAM=`echo $1 | awk -F= '{print $1}'`
+	VALUE=`echo $1 | awk -F= '{print $2}'`
+	case $PARAM in
+		-t | --tab)
+			NEWTAB=1
```
Comment (twolfson) [inline iTerm.sh]: We should keep the bash variable consistent with the option. However, `TAB=1` would be confusing for a developer as it's ambiguous. Can we move back to `--open-in-tab` and make the variable `OPEN_IN_TAB`?

```
--open-in-tab)
    OPEN_IN_TAB=1
```

**Comment by vitorhsb** [comment]
@twolfson A new commit has been made to reflect your comments.

File: iTerm.sh [hunk 6]
```diff
@@ -2,6 +2,18 @@
 
 CD_CMD="cd "\\\"$(pwd)\\\"" && clear"
 VERSION=$(sw_vers -productVersion)
+OPEN_IN_TAB=0
+
+while [ "$1" != "" ]; do
+	PARAM=`echo $1 | awk -F= '{print $1}'`
+	VALUE=`echo $1 | awk -F= '{print $2}'`
```
Comment (twolfson) [inline iTerm.sh]: These `echo` and `awk` statements don't seem practical. Can we replace them with assignments?

```bash
PARAM="$1"
VALUE="$2"
```

File: iTerm.sh [hunk 7]
```diff
@@ -2,6 +2,18 @@
...
+	case $PARAM in
```
Comment (twolfson) [inline iTerm.sh]: Please add quotes to any `bash` variable. Otherwise, we might encounter a space in the variable which will cause issues:

```bash
case "$PARAM" in
```

Comment (vitorhsb) [inline iTerm.sh]: Done.

File: iTerm.sh (hunk 6 above)
Comment (vitorhsb) [inline iTerm.sh]: Done.

**Comment by twolfson** [comment]
Alright, looks good to me. I will open up a secondary PR with README and messages for wbond to review by the end of this weekend. Thanks for the quick replies and updates =)

**Comment by vitorhsb** [comment]
Cool. Keep me up to speed if any modification is required.

**Comment by twolfson** [comment]
We have opened #85 to get this landed in `master` =)

**Comment by twolfson** [comment]
This has been landed/released in `1.8.0`. Thanks for all your help =)

**Comment by mbigras** [comment]
just a heads up for anyone reading this, I'm pretty sure that the parameter that is working for some people (just not me at the moment 😞 ) is `--open-new-tab` and not `--openNewTab`
